Random failing org.eclipse.ant.tests.ui.AntUtilTests.testGetIncludeTargetsPerformance on MacOS

`AntUtilTests.testGetIncludeTargetsPerformance` takes more than the expected maximum of 10 seconds to finish in some of the recent MacOS builds. As far as I can see there were no recent changes in the productive code, so failures may be due to too slow build infrastructure?

```
java.lang.AssertionError: Expecting a duration < 10000, but we have 12125ms
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.eclipse.ant.tests.ui.AntUtilTests.testGetIncludeTargetsPerformance(AntUtilTests.java:221)
```

See:
https://github.com/eclipse-platform/eclipse.platform/actions/runs/4621006483/jobs/8171898101
https://github.com/eclipse-platform/eclipse.platform/actions/runs/4620696572/jobs/8171161377
https://github.com/eclipse-platform/eclipse.platform/actions/runs/4616360586/jobs/8161276288

--- a/ant/org.eclipse.ant.tests.ui/Ant Tests/org/eclipse/ant/tests/ui/AntUtilTests.java
+++ b/ant/org.eclipse.ant.tests.ui/Ant Tests/org/eclipse/ant/tests/ui/AntUtilTests.java
@@ -15,6 +15,7 @@
 package org.eclipse.ant.tests.ui;
 
 import java.io.File;
+import java.lang.management.ManagementFactory;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -200,11 +201,11 @@ public class AntUtilTests extends AbstractAntUITest {
 		 * is not set), but every include-file now includes the "big build.xml"
 		 */
 		String buildFileName = "bug412809/performance/buildFileHierarchical"; //$NON-NLS-1$
-		long startTime = System.currentTimeMillis();
+		long startTimeInNanoseconds = ManagementFactory.getThreadMXBean().getCurrentThreadCpuTime();
 		File buildFile = getBuildFile(buildFileName + ".xml"); //$NON-NLS-1$
 		IAntModel model = AntUtil.getAntModel(buildFile.getAbsolutePath(), false, true, true);
 		AntProjectNode project = model.getProjectNode();
-		long endTime = System.currentTimeMillis();
+		long endTimeInNanoseconds = ManagementFactory.getThreadMXBean().getCurrentThreadCpuTime();
 
 		Assert.assertNotNull(project);
 		/*
@@ -214,11 +215,13 @@ public class AntUtilTests extends AbstractAntUITest {
 		 *
 		 * - 10s elsewhere
 		 */
-		long duration = endTime - startTime;
+
+		long durationInMilliseconds = (endTimeInNanoseconds - startTimeInNanoseconds) / 1_000_000;
 		// Change this value if it does not fit the performance needs
 		long maxDuration = this.getExecutionTresholdIncludeTask();
 
-		Assert.assertTrue("Expecting a duration < " + maxDuration + ", but we have " + duration + "ms", duration < maxDuration); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+		Assert.assertTrue("Expecting a duration < " + maxDuration + ", but we have " + durationInMilliseconds + "ms", //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+				durationInMilliseconds < maxDuration);
 		// Test the rest
 		AntTargetNode[] targets = getAntTargetNodesOfBuildFile(buildFileName);
 

